### PR TITLE
Upgrade ruby, clickhouse

### PR DESCRIPTION
**Pull request by DeathBorn**
@vinted/sre4 
@ernestas-poskus 
This PR introduces major changes
* add rockylinux support
* latest stable clickhouse support (previous installation method is deprecated by altinity)
* refactored zookeeper installation for tests to be more streamlined.
* moving to `cinc`
* upgrading in general
* remove ubuntu/debian support

### CODE REVIEW

**Review by vbalys — APPROVED** [review]
:+1:

**Review by aliulis — APPROVED** [review]

File: libraries/clickhouse_base.rb [hunk 1]
```diff
@@ -47,11 +47,7 @@ def action_delete
       end
 
       def rhel_family?
-        %w[rhel centos].include?(node['platform'])
-      end
-
-      def debian_family?
-        %w[debian ubuntu].include?(node['platform'])
+        %w[rhel centos rocky].include?(node['platform'])
```
Comment (vbalys) [inline libraries/clickhouse_base.rb]: I think there is no such platform as `rhel`. Likely `redhat`: https://github.com/chef/chef/blob/61a11902ab814aad3625eb4da7e3345d63ee7c09/chef-utils/lib/chef-utils/dsl/platform.rb#L104

File: libraries/clickhouse_base.rb [hunk 2]
```diff
@@ -75,7 +71,7 @@ def cpu_sse4_flag_present?
       end
 
       def platform_supported?
-        return if %w[redhat centos debian ubuntu].include?(node['platform'])
+        return if %w[redhat centos rocky debian ubuntu].include?(node['platform'])
```
Comment (vbalys) [inline libraries/clickhouse_base.rb:74]: In this place, `redhat`, not `rhel` - :+1:

File: libraries/clickhouse_base.rb [hunk 3]
```diff
@@ -123,20 +119,17 @@ def install_dependencies
       # rubocop:disable Metrics/MethodLength
       def install_clickhouse_repository
         case node['platform']
-        when 'rhel', 'centos'
```
Comment (vbalys) [inline libraries/clickhouse_base.rb:126]: Again, `rhel` :-)

File: libraries/clickhouse_base.rb [hunk 4]
```diff
@@ -123,20 +119,17 @@ def install_dependencies
...
-        when 'ubuntu', 'debian'
-          execute 'apt_update_for_chef_clickhouse' do
-            command %(apt-get update)
-            action :nothing
-          end
-
-          execute 'curl -s https://packagecloud.io/install/repositories/Altinity/clickhouse/script.deb.sh | sudo bash' do
-            creates '/etc/apt/sources.list.d/Altinity_clickhouse.list'
-            notifies :run, 'execute[apt_update_for_chef_clickhouse]', :immediate
+        when 'rhel', 'centos', 'rocky'
```
Comment (vbalys) [inline libraries/clickhouse_base.rb]: Here, too.

File: test/integration/default/helpers_spec.rb [hunk 5]
```diff
@@ -1,7 +1,3 @@
 def rhel_family?
-  %w[rhel centos].include?(os[:platform])
-end
-
-def debian_family?
-  %w[debian ubuntu].include?(os[:platform])
+  %w[rhel centos rocky].include?(os[:platform])
```
Comment (vbalys) [inline test/integration/default/helpers_spec.rb]: And here, too.

**Pull request merged**